src/PostgresOutbox/Serialization/ITypeResolver.cs

```diff
@@ -0,0 +1,32 @@
+using System.Text.Json.Serialization;
+using System.Text.Json.Serialization.Metadata;
+
+namespace PostgresOutbox.Serialization;
+
+public interface ITypeResolver
+{
+    Type Resolve(string value);
+    (string, JsonTypeInfo) Resolve(Type type);
+    JsonSerializerContext SerializationContext { get; }
+}
+
+public class TypeResolver(JsonSerializerContext serializationContext, INamingPolicy? namingPolicy=default): ITypeResolver
+{
+    public JsonSerializerContext SerializationContext { get; } = serializationContext;
+    private static readonly Dictionary<string, Type> TypeDictionary = [];
```

Comment:
**QUESTION:** This assumes that there won't be multithreaded registration to the type resolution? If the answer is yes, then that sounds fine. Otherwise, I'd recommend using ConcurrentDictionary, or better `ImHashMap` from [ImTools](https://github.com/dadhi/ImTools).

Reply:
ConcurrentDictionary should be fair for now

Reply:
Added https://github.com/oskardudycz/PostgresOutboxPatternWithCDC.NET/issues/11